### DISCUSSION TITLE
Is Duckdb suitable for storing log,metrics,traces?

### DISCUSSION THREAD

**Discussion by microhardsmith** [duckdb/duckdb#8521] [Q&A]
Hi there, I am writing a simple microservice framework on my own, and I have an idea about using DuckDB for application observability, and I want to know if it's suitable for this situation.

I want to store all the logs, metrics, traces in DuckDB using parquet format, and ship the data to Amazan S3 or other compatible object-storage periodically , pull the data from S3 when searching for records, I want to know if DuckDB is suitable for this use case (high frequency writing, fixed table, periodically removal, resonable query performance)

**Comment by mikemdr1** [comment]
Hi there, the only comment I have is currently DuckDB has no feature to vaccum removal data.

For example, suppose you are storing data for one month and you duckdb file has 5GB of data.
Then you ship the data to other place and run a DELETE clause for remove transfered records.
Your DuckDB will continue using 5GB in your disk. 

It's a well known issue that currently is on the roadmap to be solved.
In that case, DuckDB devs suggest to export whole database, delete DuckDB file and then create a new DuckDB file importing the exported database.
You can create a function that does that easily, it's fast and not complicated at all, but it's something to take into account.

I'm not familiar with databases for logs/metrics, but if you are exploring try TimescaleDB.
It's PostgreSQL based, has a Docker Image and has a lot of time-series functions to manipulate information

Hope this helps to take a good decision.
